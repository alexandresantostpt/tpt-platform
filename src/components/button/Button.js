import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { colors } from '@helpers/colors'

const StyledButton = styled.button`
    background: ${({ fill, theme }) => (fill ? theme.primaryDark : colors.white)};
    border: 1px solid ${({ theme, disabled }) => (disabled ? colors.lightBlueGray : theme.primaryDark)};
    border-radius: 24px;
    color: ${({ theme, disabled, fill }) => {
        if (fill) {
            return colors.white
        }
        return disabled ? colors.lightBlueGray : theme.primaryDark
    }};
    cursor: ${({ disabled }) => !disabled && 'pointer'};
    display: flex;
    font-size: 0.9rem;
    font-weight: bold;
    height: 2.3rem;
    justify-content: center;
    padding: 0.6875rem 1rem;
    text-align: center;
    text-transform: uppercase;
    white-space: nowrap;

    &:focus {
        outline: none;
    }

    &:hover {
        background: ${({ theme, disabled }) => !disabled && theme.primaryDark};
        border: ${({ disabled }) => !disabled && `1px solid ${colors.white}`};
        color: ${({ disabled }) => !disabled && colors.white};
    }
`

const Button = ({ disabled, fill, onClick, text }) => (
    <StyledButton disabled={disabled} fill={fill} onClick={onClick}>
        {text}
    </StyledButton>
)

Button.defaultProps = {
    disabled: false,
    fill: false
}

Button.propTypes = {
    disabled: PropTypes.bool,
    fill: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
    text: PropTypes.string.isRequired
}

export default Button
