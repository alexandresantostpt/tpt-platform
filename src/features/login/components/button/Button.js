import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { colors } from '@helpers/colors'

const StyledButton = styled.button`
    background: linear-gradient(-225deg, #5271c4 0%, #b19fff 48%, #eca1fe 100%);
    border-radius: 30px;
    border-style: none;
    box-sizing: border-box;
    color: ${colors.white};
    cursor: pointer;
    font-size: 1.125rem;
    font-weight: bold;
    margin-bottom: 0.8rem;
    margin-top: 2rem;
    padding: 0.8rem;
    text-transform: uppercase;
    width: 70%;

    &:focus {
        outline: none;
    }
`

const Button = ({ disabled, text, type }) => (
    <StyledButton disabled={disabled} type={type}>
        {text}
    </StyledButton>
)

Button.defaultProps = {
    disabled: false
}

Button.propTypes = {
    disabled: PropTypes.bool,
    text: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired
}

export default Button
