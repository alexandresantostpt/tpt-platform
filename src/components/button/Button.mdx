---
menu: Button
name: Button
route: /button
---
import { Playground, PropsTable } from 'docz'

import Button from './Button'

import '@cfg/scripts'
import '@cfg/style'

# Button

<PropsTable of={ Button }/>

## Basic Usage

```
<Button text={ TEXT }/>
```

<Playground>
    <Button text="TEXT"/>
</Playground>
