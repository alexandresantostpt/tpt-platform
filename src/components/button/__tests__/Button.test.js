import React from 'react'
import renderer from 'react-test-renderer'

import Button from '@components/button/Button'

describe('Tests for Button component', () => {
    it('Should rendering without crash', () => {
        const tree = renderer.create(<Button text="Test" />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
