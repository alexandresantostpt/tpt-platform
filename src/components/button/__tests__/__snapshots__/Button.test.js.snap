// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Button component Should rendering without crash 1`] = `
<button
  className="Button__StyledButton-sc-1fc3klx-0 bEGBgE"
  disabled={false}
  fill={false}
>
  Test
</button>
`;
